### PR TITLE
feat: Support both sync and async in ComponentDependencyMiddleware

### PR DESCRIPTION
**Pull request by TheSteveBurgess**
Hi,

PR for converting the middleware to async as mentioned here (https://github.com/EmilStenstrom/django-components/issues/464)

This is working fine for me both sync and async via Daphne.

### CODE REVIEW

**Comment by TheSteveBurgess** [comment]
Well, that looks like I've broken something :) 

Let me take a look why the tests are failing.

File: src/django_components/middleware.py [hunk 1]
```diff
@@ -33,14 +36,32 @@ class ComponentDependencyMiddleware:
...
+        if iscoroutinefunction(self.get_response):
+            markcoroutinefunction(self)
+
     def __call__(self, request: HttpRequest) -> HttpResponseBase:
+
+        if iscoroutinefunction(self):
+            return self.__acall__(request)
+
         response = self.get_response(request)
+        response = self.process_response(request)
```
Comment (JuroOravec) [inline src/django_components/middleware.py:48]: Ah, maybe the issue is that we pass `request` here instead of `response`?

**Comment by JuroOravec** [comment]
@TheSteveBurgess thanks a lot for the PR! 🚀 It looks great! There's one typo, let's see if things work after that :)

File: src/django_components/middleware.py (hunk 1 above)
Comment (TheSteveBurgess) [reply src/django_components/middleware.py:48]: That'd do it :) Hopefully it's working with that change

File: src/django_components/middleware.py [hunk 2]
```diff
@@ -33,14 +36,32 @@ class ComponentDependencyMiddleware:
...
     def __call__(self, request: HttpRequest) -> HttpResponseBase:
+
+        if iscoroutinefunction(self):
+            return self.__acall__(request)
+
         response = self.get_response(request)
+        response = self.process_response(response)
+        return response
+
+    async def __acall__(self, request):
```
Comment (JuroOravec) [inline src/django_components/middleware.py:51]:
```suggestion
    async def __acall__(self, request: HttpRequest) -> HttpResponseBase:
```

File: src/django_components/middleware.py [hunk 3]
```diff
@@ -33,14 +36,32 @@ class ComponentDependencyMiddleware:
...
+
+        response = await self.get_response(request)
+        response = self.process_response(response)
+        return response
+
+    def process_response(self, response):
```
Comment (JuroOravec) [inline src/django_components/middleware.py:57]:
```suggestion
    def process_response(self, response: HttpResponse) -> HttpResponse:
```

**Comment by TheSteveBurgess** [comment]
Amazing! Thanks for that :)

**Comment by JuroOravec** [comment]
Merged! @TheSteveBurgess thanks a lot again :) I plan to release this as part of v0.70 together with https://github.com/EmilStenstrom/django-components/pull/465 later today

**Pull request merged**